action/images.py

```diff
@@ -34,6 +38,31 @@ class shared_directories_action:
 shared_directories_actions: list[shared_directories_action] = []
 
 
+def docker_image_parse(image: str) -> tuple[str]:
+
+    class LoopMatch(object):
+        def __init__(self, image_string: str):
+            self.image_string = image_string
+
+        def __getitem__(self, nr: int):
+            return self.result.group(nr)
+
+        def match(self, regexp: str) -> bool:
+            self.result = re_match(regexp, self.image_string)
+            return self.result is not None
+
+    tp = LoopMatch(image)  # try pattern
```

Comment:
What is the point of this class? Why not just use `re` stuff directly?

Reply:
I did this because python do not support assignment in the if statement. The code looks cleaner for me without breaks for assignments in this if chain.

Reply:
Depends on your definition of clean.

First time readers of this code are most likely confused by this class, and they have to spend time on figuring out what's going on instead of just seeing a few `if`s that make use of a popular library.

This structure also breaks the flow of the code, as when you look at the `if`s, you have to go back to the class to understand it, and then go back to the `if`s again.

Repetition is not always bad. In this case I'd go with simpler code, as it's immediately understandable.

Reply:
Done